tests/platform_tests/sfp/test_sfputil.py

```diff
@@ -30,6 +44,183 @@
+                                                                  "HDEL",
+                                                                  logical_intf,
+                                                                  "")
+        self.db_cmd_dom_polling_get = db_cmd_dom_polling.format(self.namespace_cmd_opt,
+                                                                "HGET",
+                                                                logical_intf,
+                                                                "")
+        self.orig_dom_polling_value = None
+
+    def disable(self):
+        """
+        Disable a logical interface by doing below:
+            * Disable DOM polling
+            * Shutdown port
+            * Check and save lpmode
+        """
+        if not self.skip_dom_polling_handle:
+            orig_dom_get_result = self.duthost.command(self.db_cmd_dom_polling_get)
+            if orig_dom_get_result["stdout"] in DOM_POLLING_CONFIG_VALUES:
+                self.orig_dom_polling_value = orig_dom_get_result["stdout"]
+            logging.info("Disable DOM polling to avoid race condition during sfp reset"
+                         " for {}".format(self.logical_intf))
+            disable_dom_result = self.duthost.command(self.cmd_disable_dom)
+            assert disable_dom_result["rc"] == 0, \
+                "Disable DOM polling failed for {}".format(self.logical_intf)
+            time.sleep(self.wait_after_dom_config)
+
+        # It's needed to shutdown ports before reset and startup ports after reset,
+        # to get config/state machine/etc replayed, so that the modules can be fully
+        # restored.
+        logging.info("Shutdown {} before sfp reset".format(self.logical_intf))
+        shutdown_result = self.duthost.command(self.cmd_down)
+        assert shutdown_result["rc"] == 0, "Shutdown {} failed".format(self.logical_intf)
+        assert check_interface_status(self.duthost, [self.logical_intf], expect_up=False)
+
+        if not self.skip_lpmode_handle:
+            logging.info("Check output of '{}' before sfp reset".format(self.cmd_sfp_show_lpmode))
+            lpmode_show = self.duthost.command(self.cmd_sfp_show_lpmode)
+            self.original_lpmode_state = \
+                parse_output(lpmode_show["stdout_lines"][2:])[self.logical_intf].lower()
+
+    def restore(self):
+        """
+        Restore a logical interface from disabled state by doing below:
+            * Check and restore lpmode if needed
+            * Startup port
+            * Enable DOM polling
+        """
+        if not self.skip_lpmode_handle:
+            # Restore lpmode if needed, because sfp reset will reset lpmode to default
+            logging.info("Check output of '{}' after sfp reset".format(self.cmd_sfp_show_lpmode))
+            lpmode_show = self.duthost.command(self.cmd_sfp_show_lpmode)
+            after_lpmode_state = \
```

Comment:
why do we need recover LPM here? I think LPM should be handled by xcvrd.

Reply:
For non-cmis module, xcvrd doesn't handle LPM, which might need to be restored after sfputil reset